### DISCUSSION TITLE
How to set CORS for deployment URLs to use main domain API?

### DISCUSSION THREAD

**Discussion by jinsley8** [vercel/community#1551] [Help] [answered]
My Next.js app is fetching from the API routes using relative URLs - `/api/example`

When I deploy, the deployment URL gets a CORS error trying to fetch from the API because Vercel appends the main domain to the API routes.

Whtas the best way to set it up so any of the deployment URLs can use the main API routes?

Error:
```
Access to fetch at 'https://domain.vercel.app/api/example' from origin 'https://domain-dorq3glj0-dev.vercel.app' has been 
blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' 
header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to 
fetch the resource with CORS disabled.
```

**Comment by amyegan** [comment, marked as answer]
Hi @jinsley8! We have a guide that shows a few ways to add CORS headers to solve this problem: https://vercel.com/guides/how-to-enable-cors

**Comment by eybel** [comment]
Hi there, I have the same issue but my backend app is an express.js App in AWS. When I do a POST (upload) function I get CORS and then I checked my AWS logs and I could see that instead of being my main domain the ORIGIN (the one I set up as valid origin) the request came from something like => **"https://my-domain-6axshagkz-eybel.vercel.app"** instead of being "https://www.my-domain.com"

Why isn't coming from main domain? This seems to be the preview domain. I could fix it with a REGEX or something but wanted to ask first.

AWS logs hitting the CORS policy:
https://ibb.co/pLZDKVW

Thank you

**Reply by stithwilliam** [reply]
I have this same problem.
